sdk/metric/meter.go

```diff
@@ -83,6 +83,19 @@ func (r *meterRegistry) Get(s instrumentation.Scope) *meter {
 type meter struct {
 	instrumentation.Scope
 
+	// aggregatorCache* ensures no duplicate Aggregators are created for the
+	// same instrument within the scope of all instruments this meter owns.
+	//
+	// Duplicate instrument creations for different number types are identified
+	// in the viewCache. Since the conflict is "resolvable", a valid aggregator
+	// still needs to be returned when this occurs. Therefore, instruments of
+	// different numbers are not tracked with the same cache.
+	aggregatorCacheInt64   cache[instrumentID, aggVal[int64]]
+	aggregatorCacheFloat64 cache[instrumentID, aggVal[float64]]
+	// viewCache ensures instrument conflicts this meter is asked to create are
+	// logged to the user.
+	viewCache cache[string, instrumentID]
```

Comment:
Is there a reason to have these as the raw aggregatorsCache and view caches, which we use to create InstrumentChaches in the sync/async methods, over just storing the instrument caches here?

Reply:
Yeah, I was thinking how to stage this change with the follow on ones related to caching that would unify this. Given you're asking, it make sense that I should include the field differently here. I'll update.